gson/src/org/immutables/gson/stream/JsonParserReader.java

```diff
@@ -141,7 +144,7 @@ public String nextName() throws IOException {
   public String nextString() throws IOException {
     requirePeek();
     if (!isLenient()) {
-      expect(VALUE_STRING);
+      expectOneOf(VALUE_STRING, VALUE_NUMBER_FLOAT, VALUE_NUMBER_INT);
```

Comment:
As far as I see it, current behavior of our subclass breaks things that Gson "expects" to do with all those `LazilyParsedNumber`. I rightfully dislike the change above, but not because a change is not needed, we just doing(or already did) something wrong and we need to revise the check. Let me merge the PR and edit the fragment in question.